curvefs/src/client/fuse_s3_client.cpp

```diff
@@ -76,10 +74,270 @@ CURVEFS_ERROR FuseS3Client::Init(const FuseClientOption &option) {
                                nullptr, true);
     }
 
+    bgFetchStop_.store(false, std::memory_order_release);
+    bgFetchThread_ = Thread(&FuseS3Client::BackGroundFetch, this);
+    GetTaskFetchPool();
     return ret;
 }
 
+void FuseS3Client::BackGroundFetch() {
+    while (!bgFetchStop_.load(std::memory_order_acquire)) {
+        LOG_EVERY_N(WARNING, 100)
+            << "fetch thread start.";
+        if (hasWarmUpTask()) {  // new warmup task
+            WarmUpFileContext_t warmUpFile;
+            GetWarmUpFile(&warmUpFile);
+            VLOG(9) << " len is: " << warmUpFile.fileLen
+                << "ino is: " << warmUpFile.inode;
+            struct fuse_file_info fi;
+            fi.flags &= ~O_DIRECT;
+            size_t rSize = 0;
+            std::unique_ptr<char[]> data(new char[warmUpFile.fileLen+1]);
+            std::memset(data.get(), 0, warmUpFile.fileLen);
```

Comment:
Please use a function directly to process string

Reply:
got it